tests/ui/infinite_loops.rs

```diff
@@ -363,4 +368,18 @@ fn inf_loop_in_res() -> Result<(), i32> {
     })
 }
 
+with_span! { span
+    fn no_loop() {}
+}
+
```

Comment:
Sorry, one more thing - can you add a test where the `with_span!` macro only has the loop, and not the whole function?
```rs
fn foo() {
  with_span! { span
    loop {}
  }
}
```

Reply:
I just pushed the change, but then realized it might not be exactly what you are requesting. I included an empty function to prevent the clippy::empty_loop lint from flagging.  Is this ok, or should I add it exactly as your example, and add #[allow(clippy::empty_loop)]?

Reply:
@CBSpeir  Ah, I didn't see your comment, sorry. It doesn't really matter too much. We just don't want warnings from multiple lints in the same file, so either allowing the other lint or working around it like you did is fine, as long as the test still "works correctly". In this case, the lint `infinite_loop` understands function calls so that works too.